### PR TITLE
Reduce duplication in ci workflow file

### PR DESCRIPTION
**Pull request by nineinchnick**
## Description

Extract processing test results to an action to reuse it in multiple jobs.

Also update most actions to remove warnings about Node.js version deprecation that show up in the run summary view, obstructing other annotations.

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Review by wendigo — APPROVED** [review]

File: .github/actions/process-test-results/action.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    # Always upload the test report for the annotate.yml workflow,
+    # but only the single XML file to keep the artifact small
+    with:
+      # Name prefix is checked in the `Annotate checks` workflow
+      name: test report ${{ inputs.artifact-name }}
+      if-no-files-found: 'ignore'
+      path: |
+        **/surefire-reports/TEST-*.xml
+        testing/trino-product-tests/target/reports/**/testng-results.xml
+      retention-days: 30
```
Comment (wendigo) [inline .github/actions/process-test-results/action.yml:42]: why 30 days?

Comment (nineinchnick) [reply .github/actions/process-test-results/action.yml]: We used 5 days before, but I've chosen this value arbitrarily: https://github.com/trinodb/trino/pull/10137/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03f

I've set it to 5 now.

Note that the default is 90 days (max allowed value): https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration#artifact-and-log-retention-policy

Comment (hashhar) [reply .github/actions/process-test-results/action.yml]: this doesn't impact CI logs retention, correct?

**Review by hashhar — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -778,34 +724,26 @@ jobs:
...
+              || steps.tests-azure.outcome == 'failure'
+              || steps.tests-gcs.outcome == 'failure'
+              || steps.tests-delta.outcome == 'failure'
+              || steps.tests-memsql.outcome == 'failure'
+              || steps.tests-bq.outcome == 'failure'
+              || steps.tests-bq-ci.outcome == 'failure'
+              || steps.tests-bq-smoke.outcome == 'failure'
+              || steps.tests-iceberg.outcome == 'failure'
+              || steps.tests-redshift.outcome == 'failure'
+            }}
```
Comment (findepi) [inline .github/workflows/ci.yml:745]: How we gonna keep this up to date?
Maybe we can have a test that takes ci.yml file and asserts that (1) each test task has ID and (2) all are mentioned here?

**Pull request merged**